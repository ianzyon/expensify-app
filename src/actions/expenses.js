import database from '../firebase/firebase'

// Expenses Action Generators:
// ADD_EXPENSE
export const addExpense = (expense) => {
    return {
        type: 'ADD_EXPENSE',   
        expense
    };
};
// causando o dispatch a disparar uma nova funcao gracas ao thunk
export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {description = '', note = '', amount = 0, createdAt = 0 } = expenseData;// outra maneira de definir o default e ao mesmo tempo setar o conteudo de expenseData para as chaves

        const expense = { description, note, amount, createdAt }
        database.ref('expenses').push(expense) // empurra para o firebase os dados de expense
        .then((ref)=>{ // depois de setar no banco de dados, o state é alterado com um call para dispatch addExpense, isso é feito para usar o id gerado pelo firebase ref.key
            dispatch(addExpense({
                id: ref.key,
                ...expense
                }))
        }).catch((e)=>{console.log(e);});
    };
};
// REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => {
    return {
        type: 'REMOVE_EXPENSE',
        id
    };
};
// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});