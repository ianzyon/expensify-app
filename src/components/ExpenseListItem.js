import React from 'react';
// importar uma action generator de REMOVE
import { Link } from 'react-router-dom';

// stateless funcitional component
// description, amount, createdAt
const ExpenseListItem = ({id,description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>Amount: {amount} - Created at: {createdAt}</p>
        
    </div>
)

export default (ExpenseListItem);