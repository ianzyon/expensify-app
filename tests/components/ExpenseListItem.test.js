import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ExpenseListItem from '../../src/components/ExpenseListItem';
import expenses from '../fixtures/expense-state';

// populated
test('should render ExpenseListItem with a expense', ()=>{
    const wrapper = shallow(<ExpenseListItem key={expenses[1].id} {...expenses[1]}/>);

    expect(toJSON(wrapper)).toMatchSnapshot();
});