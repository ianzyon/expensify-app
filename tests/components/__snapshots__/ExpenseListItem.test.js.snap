// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with a expense 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    Amount: 
    19500
     - Created at: 
    -345600000
  </p>
</div>
`;
